File: GlassDoor/ViewModels/PostJobDto.cs
﻿using DAL.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace GlassDoor.ViewModels
{
    public class PostJobDto
    {
        public int? Id { get; set; }
        [Required]
        public string Title { get; set; }
        public int JobTypeId { get; set; }
        public int? NumberOfVacancies { get; set; }
        public int CityId { get; set; }
        public int CountryId { get; set; }
        public bool? IsOpen { get; set; }
        public CompanyJobDetailsDto JobDetails { get; set; }
        public ICollection<CompanyJobSkillDto> Skills { get; set; }
    }
}

File: GlassDoor/ClientApp/src/app/modules/company/models/post-job-dto.ts
import { publish } from "rxjs/operators";
import { JobDetailsDto } from "./job-details-dto";
import { JobSkill } from "./job-skill";

export class PostJobDto {
  constructor(public title:  string, public jobTypeId: number, public cityId: number, public countryId: number, public numberOfVacancies: number, public jobDetails: JobDetailsDto, public skills: JobSkill[],public id?:number)
    {
        
  
        }
      }
      

